src/llm/continuous_batching/servable_initializer.cpp

```diff
@@ -0,0 +1,196 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <openvino/genai/continuous_batching_pipeline.hpp>
+#include <openvino/openvino.hpp>
+#include <spdlog/spdlog.h>
+
+#include "../../json_parser.hpp"
+#include "../../logging.hpp"
+#include "../../status.hpp"
+#include "servable.hpp"
+
+#pragma warning(push)
+#pragma warning(disable : 4005 4309 6001 6385 6386 6326 6011 4005)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#pragma GCC diagnostic pop
+#pragma warning(pop)
+
+#include "../../mediapipe_internal/mediapipe_utils.hpp"
+#include "servable_initializer.hpp"
+#include "src/llm/text_processor.hpp"
+
+namespace ovms {
+
+ov::genai::SchedulerConfig ContinuousBatchingServableInitializer::prepareDraftPipelineSchedulerConfigExperimental(const mediapipe::LLMCalculatorOptions_PipelineConfig& draftPipelineConfig) {
+    ov::genai::SchedulerConfig config;
+    config.max_num_batched_tokens = draftPipelineConfig.max_num_batched_tokens();
+    config.cache_size = draftPipelineConfig.cache_size();
+    config.dynamic_split_fuse = draftPipelineConfig.dynamic_split_fuse();
+    config.max_num_seqs = draftPipelineConfig.max_num_seqs();
+    config.enable_prefix_caching = draftPipelineConfig.enable_prefix_caching();
+    return config;
+}
+
+ov::genai::SchedulerConfig ContinuousBatchingServableInitializer::prepareDraftPipelineSchedulerConfig(const mediapipe::LLMCalculatorOptions& nodeOptions) {
+    ov::genai::SchedulerConfig config;
+    config.max_num_batched_tokens = nodeOptions.has_draft_max_num_batched_tokens() ? nodeOptions.draft_max_num_batched_tokens() : nodeOptions.max_num_batched_tokens();
+    config.cache_size = nodeOptions.has_draft_cache_size() ? nodeOptions.draft_cache_size() : nodeOptions.cache_size();
+    config.dynamic_split_fuse = nodeOptions.has_draft_dynamic_split_fuse() ? nodeOptions.draft_dynamic_split_fuse() : nodeOptions.dynamic_split_fuse();
+    config.max_num_seqs = nodeOptions.has_draft_max_num_seqs() ? nodeOptions.draft_max_num_seqs() : nodeOptions.max_num_seqs();
+    config.enable_prefix_caching = nodeOptions.enable_prefix_caching();
+    return config;
+}
+
+Status ContinuousBatchingServableInitializer::initializeExperimental(std::shared_ptr<GenAiServable>& servable, const mediapipe::LLMCalculatorOptions& nodeOptions, std::string graphPath) {
```

Comment:
this is actual implementation of initializer - `ContinuousBatchingServableInitializer`, why do we take base class `std::shared_ptr<GenAiServable>& servable` as parameter? we could already use actual derived class to avoid cast

Reply:
`initialize` is part of the interface and it's signature must be the same for all pipelines. 
I did it that way to have a unified path and clear convention of initialization for GenAI pipelines.